scalding_typed/src/main/scala/com/matthewrathbone/scalding/Main.scala

```diff
@@ -0,0 +1,34 @@
+package main.scala.com.matthewrathbone.scalding
+
+import com.twitter.scalding._
+import TDsl._
+import cascading.tuple.TupleEntry
+
+
+class Main ( args: Args ) extends Job( args ) {
+
+	val output = args( "output" )
+	
+	val input1 : TypedPipe[String] = TypedPipe.from(TextLine(args( "input1" )))
+	val input2 : TypedPipe[String] = TypedPipe.from(TextLine(args( "input2" )))
+	
+	val usersInput : TypedPipe[(String, String, String, String)] = input1.map{ s:String =>
+      val split = s.split("\t")
+      (split( 0 ), split( 1 ), split( 2 ), split( 3 ))
+	}
+	val group1 = usersInput.groupBy { case (id, email, language, location) => id }
+
+	val transactionsInput : TypedPipe[(String, String, String, String, String)] = input2.map{ s:String =>
```

Comment:
Let's use a case class here instead of a Tuple -- that way we properly show off how useful the Typed API can be.

eg:

```scala

case class Transaction(id: Long, productId: Long, userId: Long, purchaseAmount: Double, itemDescription: String)
case class User(...)

Then we can have a `TypedPipe[Transaction]`


```